src/NBitcoin/TransactionBuilder.cs

```diff
@@ -139,12 +139,36 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
                             }
                         }
 
-                        return result;
+                        if (minTotal != null)
+                        {
+                            total = minTotal;
+                        }
+                        break;
                     }
                 }
             }
+
             if (total.CompareTo(target) == -1)
                 return null;
+
+            // optimize the set of used coins, removing unnecessary small inputs
+            List<ICoin> sortedUsedCoins = result.OrderBy(c => c.Amount).ToList();
+            IMoney excess = total.Sub(target);
+            for (int i = 0; i < sortedUsedCoins.Count; i++)
+            {
+                // if the smallest coin in the set is smaller or equal to the difference between the excess of
```

Comment:
Do you take the fee in consideration here?

Reply:
The TB can do many things hence the complexity.

If fee is sort of already in the target then ok its fine, though the fee can change depending on how coin selection finds outputs.

I guess as long as there is a minimal fee that is applied the end result may at worst be a delay in getting in to a block